### DISCUSSION TITLE
Question: the reason for bypassing auto-detection mode

### DISCUSSION THREAD

**Discussion by kostyaplis** [ansible/ansible-lint#2427] [Q&A] [answered]
Hypothetical scenario:
In huge roles/playbooks repository we would like to perform a regression tests only on modified files.
So we'd pass modified files to ansible-lint as such:
```
ansible-lint playbook.yml roles/some_role/tasks/task.yml just_a_yaml_file.yml
``` 
as we are passing lintable args, the [following logic](https://github.com/ansible/ansible-lint/blame/main/src/ansiblelint/utils.py#L834-L841) got triggered and `yaml` kinds are getting replaced by `playbook` kind
That leads to unexpected lint errors, so we cannot use ansible-lint to perform `yaml` checks.

@ssbarnea could you advise on a reason for this logic, please.
What would be the best way to fix it? Remove this logic (it case it's obsolete) or introduce a cli arg to enforce auto-detection?

**Comment by ssbarnea** [comment, marked as answer]
Linting diffs does not work, mainly because a change make to one file can cause a breakage in a totally different file, one that is not touched. You are not the first, or the last to ask this question, sadly, the partial linting works only for playbook file type, so we did not bother to implement it.

Regarding auto-detection of file type, you can customize the file patterns to match your needs. We also plan to remove the assumption that any passed file is a playbook but that counted as a major breaking change, as some people might be surprised to discover that ansible-lint passed without problems just because the yaml file was detected as not being owned by ansible (random yaml file for example).

**Reply by kostyaplis** [reply]
Completely agree with an opinion on diffs linting. We understand pros/cons, but for now diff linting is the only viable solution for us.

As for yaml->playbook assumption when lintable argument passed - is its removal planned for any particular release? 
Would it make sense to:
1. Add a deprecation notice in addition to an existing `Overriding detected file kind...` warning?
2. Add a config file property to force auto-detection and bypass overriding logic? I understand that, unlike cli args, a config file property can be added/removed with less impact on the community
I can help on that

**Reply by kostyaplis** [reply]
>  We also plan to remove the assumption that any passed file is a playbook ...
@ssbarnea do you have a milestone for this change? So I know when to come back to check...this is the only issue that requires us to build a patched version and prevents us from using upstream releases
